course_catalog/views_test.py

```diff
@@ -725,31 +726,50 @@ def test_recent_podcast_episodes_no_feature_flag(settings, client):
         assert resp.status_code == status.HTTP_403_FORBIDDEN
 
 
-def test_recent_podcast_episodes(settings, client):
-    """Recent podcast episodes API should return recent serialized podcast episodes in order of most recent first"""
-    episodes = list(
-        reversed(
-            sorted(
-                PodcastEpisodeFactory.create_batch(5),
-                key=lambda episode: (episode.last_modified, episode.id),
-            )
-        )
+def test_podcast_episodes(settings, client):
+    """
+    Podcast episodes and tecent podcast episodes APIs should return recent serialized podcast episodes in order of most recent first.
+    Recent podcast episodes API should limit to only one episode per podcast
+    """
+    latest_episodes_by_podcast = PodcastEpisodeFactory.create_batch(5)
+
+    older_episode = PodcastEpisodeFactory.create(
+        podcast=latest_episodes_by_podcast[0].podcast,
+        last_modified=latest_episodes_by_podcast[0].last_modified - timedelta(days=10),
     )
+
     # Make sure these don't get counted
     PodcastEpisodeFactory.create(published=False)
     PodcastEpisodeFactory.create(published=True, podcast__published=False)
 
     settings.FEATURES[features.PODCAST_APIS] = True
     for basename in ["recent-podcast-episodes", "podcastepisodes-list"]:
+        if basename == "recent-podcast-episodes":
+            episodes = latest_episodes_by_podcast
+            count = 5
+        else:
+            episodes = latest_episodes_by_podcast + [older_episode]
+            count = 6
+
+        episodes = list(
+            reversed(
+                sorted(
+                    episodes, key=lambda episode: (episode.last_modified, episode.id)
+                )
+            )
+        )
+
         resp = client.get(reverse(basename))
         assert resp.status_code == status.HTTP_200_OK
-        assert resp.json()["count"] == 5
+        assert resp.json()["count"] == count
 
-        assert resp.json()["results"] == [
-            {"is_favorite": False, **episode}
+        expected_results = [
```

Comment:
Once this test failed on me because the topics were in different orders.  Could modify the serializer to always return topics in a sorted order, or just pop out the topics from the JSON before making the comparison.

Reply:
I think I just need to add sort_keys=True to `dumps(`. That sorts nested keys as well